src/networkcloud/azext_networkcloud/aaz/latest/networkcloud/virtualmachine/_power_off.py

```diff
@@ -0,0 +1,326 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "networkcloud virtualmachine power-off",
+    is_experimental=True,
+)
+class PowerOff(AAZCommand):
+    """Power off the provided virtual machine.
+
+    :example: Power off virtual machine
+        az networkcloud virtualmachine power-off --resource-group "resourceGroupName" --name "virtualMachineName" --skip-shutdown "True"
+    """
+
+    _aaz_info = {
+        "version": "2022-12-12-preview",
+        "resources": [
+            ["mgmt-plane", "/subscriptions/{}/resourcegroups/{}/providers/microsoft.networkcloud/virtualmachines/{}/poweroff", "2022-12-12-preview"],
+        ]
+    }
+
+    AZ_SUPPORT_NO_WAIT = True
+
+    def _handler(self, command_args):
+        super()._handler(command_args)
+        return self.build_lro_poller(self._execute_operations, self._output)
+
+    _args_schema = None
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        if cls._args_schema is not None:
+            return cls._args_schema
+        cls._args_schema = super()._build_arguments_schema(*args, **kwargs)
+
+        # define Arg Group ""
+
+        _args_schema = cls._args_schema
+        _args_schema.resource_group = AAZResourceGroupNameArg(
+            required=True,
+        )
+        _args_schema.virtual_machine_name = AAZStrArg(
+            options=["-n", "--name", "--virtual-machine-name"],
+            help="The name of the virtual machine.",
+            required=True,
+            id_part="name",
+            fmt=AAZStrArgFormat(
+                pattern="^([a-zA-Z0-9][a-zA-Z0-9]{0,62}[a-zA-Z0-9])$",
+            ),
+        )
+
+        # define Arg Group "VirtualMachinePowerOffParameters"
+
+        _args_schema = cls._args_schema
+        _args_schema.skip_shutdown = AAZStrArg(
```

Comment:
Please verify it's string type in api

Reply:
yes, it is a string. https://github.com/Azure/azure-rest-api-specs/blame/main/specification/networkcloud/resource-manager/Microsoft.NetworkCloud/preview/2022-12-12-preview/networkcloud.json#L6653

It is enum True/False represented as string type as per API guidelines.